docs/4.0/components/forms.md

```diff
@@ -765,17 +765,45 @@ When attempting to submit, you'll see the `:invalid` and `:valid` styles applied
 // Example starter JavaScript for disabling form submissions if there are invalid fields
 (function() {
   'use strict';
+  // Generic function to manage aria-describedby attribute value on input validity update
+  // @param array of inputs to check
+  // Works only if the invalid-feedback element have an ID
+  function updateAriaDescribedAttribute(inputs) {
+    for (var i = 0; i < inputs.length; i++) {
+      // Check element validity, and set aria-described attribute
+      if (inputs[i].validity.valid === false) {
+        if (inputs[i].parentElement.getElementsByClassName('invalid-feedback').length > 0
+          && inputs[i].parentElement.getElementsByClassName('invalid-feedback')[0].id) {
+          inputs[i].setAttribute('aria-describedby', inputs[i].parentElement.getElementsByClassName('invalid-feedback')[0].id)
+          inputs[i].setAttribute('aria-invalid', true)
+        }
+      } else {
+        inputs[i].removeAttribute('aria-describedBy')
+        inputs[i].setAttribute('aria-invalid', false)
+      }
+    }
+  }
 
   window.addEventListener('load', function() {
-    var form = document.getElementById('needs-validation');
+    var form = document.getElementById('needs-validation')
     form.addEventListener('submit', function(event) {
       if (form.checkValidity() === false) {
-        event.preventDefault();
-        event.stopPropagation();
+        event.preventDefault()
+        event.stopPropagation()
+        // Check each form elements validity and add an aria-describedby to associate properly feedback with input
```

Comment:
`Check each form element's validity and set appropriate aria-invalid / aria-describedby attributes`